### DISCUSSION TITLE
Add Shader support to StyleBoxes.

### DISCUSSION THREAD

**Discussion by INdIEDeV016** [godotengine/godot-proposals#5734] [GUI]
<h2>The Problem:</h2>
As of now, I think there is no way to add a shader directly to a <code>StyleBox</code>...
<img width="153" alt="image" src="https://user-images.githubusercontent.com/80198819/200192949-e974575b-90fd-4929-98a8-6d31e987fb50.png">

...instead of adding a material to <code>Control</code> node...
<img width="235" alt="image" src="https://user-images.githubusercontent.com/80198819/200192986-16fb8014-e4cb-4556-a914-923caa2cce95.png">

...which sometimes also affects other things which are not desired (such as the shader or the material getting applied onto the text of the `Button` node as well, which is undesired)...
<img width="658" alt="image" src="https://user-images.githubusercontent.com/80198819/200193130-2c89db82-f8ac-47b8-9a83-63f51e6b049d.png">

<h3>Possible Solutions:</h3>
<ol>
<li>What I think is, if we could extend the functionality of <code>StyleBoxEmpty</code> to be able to add shaders directly as a property. This way, even if a user wants to use it as a "display nothing" <code>StyleBox</code>, it will serve the purpose as well as provide the functionality of shading.</li>

<li>Add a new <code>StyleBoxShader</code> resource extending from <code>StyleBox</code> which could serve the same purpose as above.</li>
</ol>

The ability to add shaders to `StyleBox` means that it could be reused with other `Control` nodes and thus also partly enable us to use animations in themes (See #5660).

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/3494.
